Source/WebCore/page/SpeechSynthesisClient.h

```diff
@@ -27,26 +27,15 @@
 
 #if ENABLE(SPEECH_SYNTHESIS)
 
-#include <wtf/WeakPtr.h>
-
-namespace WebCore {
-class SpeechSynthesisClientObserver;
-class SpeechSynthesisClient;
-}
-
-namespace WTF {
-template<typename T> struct IsDeprecatedWeakRefSmartPointerException;
-template<> struct IsDeprecatedWeakRefSmartPointerException<WebCore::SpeechSynthesisClientObserver> : std::true_type { };
-template<> struct IsDeprecatedWeakRefSmartPointerException<WebCore::SpeechSynthesisClient> : std::true_type { };
-}
+#include <wtf/RefCountedAndCanMakeWeakPtr.h>
 
 namespace WebCore {
 
 class PlatformSpeechSynthesisUtterance;
 class SpeechSynthesisClientObserver;
 class PlatformSpeechSynthesisVoice;
     
-class SpeechSynthesisClient : public CanMakeWeakPtr<SpeechSynthesisClient> {
+class SpeechSynthesisClient : public RefCountedAndCanMakeWeakPtr<SpeechSynthesisClient> {
```

Comment:
This is sketchy. It's OK for an abstract base class to declare a lifetime interface, but we don't want it to define the lifetime implementation, because an abstract base class has no idea what its concrete subclass is, or how it will be allocated or managed. 

So I think you want CanMakeWeakPtr, and virtual ref/deref in this class.

Reply:
(The typical failure mode for an abstract base class defining RefCounted is the case where a concrete subclass inherits from more than one abstract base class.)

Reply:
Got it--that's a good point. I did in fact encounter this exact scenario of a base class inheriting ref/deref from multiple classes in another patch. Fixed!